DashSync/shared/Models/CoinJoin/Utils/DSMasternodeGroup.h

```diff
@@ -16,12 +16,28 @@
 //
 
 #import <Foundation/Foundation.h>
+#import "DSChain.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
+@class DSCoinJoinManager;
+
 @interface DSMasternodeGroup : NSObject
 
--(BOOL)isMasternodeOrDisconnectRequested;
+@property (atomic, readonly) BOOL isRunning;
+@property (nonatomic, strong) id blocksObserver;
+@property (nonatomic, strong) DSChain *chain;
+@property (nonatomic, weak, nullable) DSCoinJoinManager *coinJoinManager;
+@property (nonatomic, strong) NSMutableSet<NSValue *> *pendingSessions;
+@property (nonatomic, strong) NSMutableDictionary *masternodeMap;
+@property (atomic, readonly) NSUInteger maxConnections;
+
+- (instancetype)initWithManager:(DSCoinJoinManager *)manager;
+
+- (void)startAsync;
+- (void)stopAsync;
+- (BOOL)isMasternodeOrDisconnectRequested;
+- (BOOL)addPendingMasternode:(UInt256)proTxHash clientSessionId:(UInt256)sessionId;
```

Comment:
In DashJ, `MasternodeGroup` is a child of `PeerGroup` which is somewhat analogous to `DSPeerManager`, but not exactly.

Given the difficulty of matching these two, I'll implement the necessary parts of `DSMasternodeGroup` step by step.